custom/enikshay/tasks.py

```diff
@@ -483,25 +484,34 @@ def get_aggregated_scores(self, latest_adherence_date, adherence_schedule_date_s
             end_date=update["aggregated_score_date_calculated"]
         )
 
+        doses_per_week = self.get_doses_per_week()
+        update['expected_doses_taken'] = int(((
+            (update['aggregated_score_date_calculated'] - adherence_schedule_date_start)).days / 7.0
+        ) * doses_per_week)
+
+        update['total_expected_doses_taken'] = self.get_total_expected_doses_taken(adherence_schedule_date_start)
+        return update
+
+    @memoized
+    def get_doses_per_week(self):
         # calculate 'expected_doses_taken' score
         dose_data = self.get_doses_data()
         adherence_schedule_id = self.episode.get_case_property('adherence_schedule_id') or DAILY_SCHEDULE_ID
         doses_per_week = dose_data.get(adherence_schedule_id)
-        if doses_per_week:
-            update['expected_doses_taken'] = int(((
-                (update['aggregated_score_date_calculated'] - adherence_schedule_date_start)).days / 7.0
-            ) * doses_per_week)
-        else:
-            update['expected_doses_taken'] = 0
+        if not doses_per_week:
             soft_assert(notify_admins=True)(
                 True,
                 "No fixture item found with schedule_id {}".format(adherence_schedule_id)
             )
-        total_expected_doses_taken = ((datetime.date.today() - adherence_schedule_date_start).days / 7.0)\
-                                     * doses_per_week
-        update['total_expected_doses_taken'] = total_expected_doses_taken
+            return 0
+        return doses_per_week
 
-        return update
+    def get_total_expected_doses_taken(self, adherence_schedule_date_start):
+        doses_per_week = self.get_doses_per_week()
+        today = self.date_today_in_india
+        total_expected_doses_taken = int(
+            round(((today - adherence_schedule_date_start).days / 7.0) * doses_per_week))
```

Comment:
this date range is also not inclusive, but maybe that's intentional

Reply:
Good question. I think this is OK as is. The task runs first thing in the morning ("midnight").
If `adherence_schedule_date_start` was yesterday, we assume that you should have only taken 1 dose at this point. i.e. `today-adherence_schedule_date_start` should = 1

Reply:
makes sense. does the other date range i pointed out need to be inclusive?

Reply:
hmm, I didn't actually change that one in this PR (just moved it). I think the different cases of `aggregated_score_date_calculated` should actually do different things here, but I'm going to check and will update in a separate PR if needbe.